matrix.js

```diff
@@ -189,9 +189,10 @@
 
     var size = i * j,
         data = new type(size),
+        isValueFn = typeof value === 'function',
         k;
     for (k = 0; k < size; k++)
-      data[k] = value;
+      data[k] = (isValueFn && value()) || value;
```

Comment:
Two issues with this approach.
1) When value() returns false, code will try to set data[k] the function instead.
2) Value function should probably take (row, col) as input to improve its usability

Summarized, better use something like this:
`      data[k] = isValueFn ? value(k / j | 0, k % j) : value;`

Note: adapted from `Matrix.prototype.map` which uses same approach for row/col computation.

Side note: for consistency of value names `fill` should probably use `r, c` as arguments: `i` and `j` are more often used as element indices, not size parameters.